Source/MQTTnet/Client/Receiving/MqttApplicationMessageReceivedEventArgs.cs

```diff
@@ -25,6 +25,7 @@ public sealed class MqttApplicationMessageReceivedEventArgs : EventArgs
             ClientId = clientId;
             ApplicationMessage = applicationMessage ?? throw new ArgumentNullException(nameof(applicationMessage));
             PublishPacket = publishPacket;
+            PacketIdentifier = publishPacket.PacketIdentifier;
```

Comment:
@chkr1011 @koepalex If `publishPacket` is `null`, this will throw. Maybe a `null` check for `publishPacket` should be added, like for `applicationMessage` in line 26?

Reply:
...or `PacketIdentifier` should not be set if `publishPacket` is `null`, leaving `PacketIdentifier` at its default value. This is the option that I would prefer, as existing code (for example unit testing code) does not need to be adapted to always supply a non-`null` `publishPacket`.